providers/asrockrack/firmware.go

```diff
@@ -74,7 +74,16 @@ func (a *ASRockRack) firmwareInstallBMC(ctx context.Context, reader io.Reader, f
 	a.log.V(2).WithValues("step", "2/4").Info("upload BMC firmware image")
 	err = a.uploadFirmware(ctx, "api/maintenance/firmware", reader, fileSize)
```

Comment:
Would prefer we identify the device model number and call the appropriate endpoint,

The model number can be obtained through this method
https://github.com/bmc-toolbox/bmclib/blob/153e11d62a71723d41a908ad362ab5e6cfc4cd54/providers/asrockrack/inventory.go#L77

Reply:
Done